### PR TITLE
Allow Kafka Sink From attribute updates

### LINKED ISSUE #571: Kafka Sink Resource: implement alter sink
https://github.com/MaterializeInc/materialize/pull/27379

### PR DESCRIPTION
**Pull request by bobbyiliev**
Fixes #571

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ services:
 
   redpanda:
     container_name: redpanda
-    image: docker.vectorized.io/vectorized/redpanda:v21.11.2
+    image: docker.vectorized.io/vectorized/redpanda:v23.3.5
```
Comment (bobbyiliev) [inline docker-compose.yml:42]: Taking the chance to upgrade the Redpanda instance.

File: pkg/resources/resource_sink_kafka.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ var sinkKafkaSchema = map[string]*schema.Schema{
 		Elem:        "from",
 		Description: "The name of the source, table or materialized view you want to send to the sink.",
 		Required:    true,
-		ForceNew:    true,
+		ForceNew:    false,
```
Comment (bobbyiliev) [inline pkg/resources/resource_sink_kafka.go:27]: By setting the `ForceNew` flag to false, we will no longer have to drop and re-create the sink if the `from` attribute changes.

File: pkg/materialize/generic.go [hunk 3]
```diff
@@ -161,3 +161,8 @@ func (b *Builder) alterDrop(name string, options []string, validate bool) error
 
 	return b.exec(query)
 }
+
+func (b *Builder) alterFrom(name string, from IdentifierSchemaStruct) error {
+	query := fmt.Sprintf(`ALTER %s %s SET FROM %s;`, b.entity, name, from.QualifiedName())
```
Comment (RobinClowers) [inline pkg/materialize/generic.go:166]: Does `QualifiedName` handle escaping the input?

**Review by RobinClowers — APPROVED** [review]

File: pkg/materialize/generic.go (hunk 3 above)
Comment (bobbyiliev) [reply pkg/materialize/generic.go:166]: Yep!

```go
func QuoteIdentifier(input string) string {
	return `"` + strings.Replace(input, `"`, `""`, -1) + `"`
}

func QualifiedName(fields ...string) string {
	var o []string
	for _, f := range fields {
		c := QuoteIdentifier(f)
		o = append(o, c)
	}

	q := strings.Join(o[:], ".")
	return q
}
```

**Pull request merged**